src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -1930,6 +1930,21 @@ public ErrorDiagnostic RuntimeValueNotAllowedInFunctionDeclaration(string? acces
                 TextSpan,
                 "BCP343",
                 $@"Using a func declaration statement requires enabling EXPERIMENTAL feature ""{nameof(ExperimentalFeaturesEnabled.UserDefinedFunctions)}"".");
+
+            public ErrorDiagnostic ExpectedAssertIdentifier() => new(
+                TextSpan,
+                "BCP344",
+                "Expected an assert identifier at this location.");
+
+            public ErrorDiagnostic AssertsUnsupported() => new(
+                TextSpan,
+                "BCP345",
+                $@"Using an assert declaration requires enabling EXPERIMENTAL feature ""{nameof(ExperimentalFeaturesEnabled.Asserts)}"".");
+
+            public ErrorDiagnostic InvalidAssertAssignment(TypeSymbol valueType) => new(
+                TextSpan,
+                "BCP346",
+                $"Value of type \"{valueType.TypeKind}\" cannot be assigned to an assert. Asserts can take values of type boolean only.");
```

Comment:
>valueType.TypeKind [](http://example.com/codeflow?start=35&length=18)

I tried this out and got this error message:
```bicep
assert a = 42
```
"Value of type "IntegerLiteral" cannot be assigned to an assert. Asserts can take values of type boolean only."

The users may not fully understand what something like `IntegerLiteral` means, so we want to print out a more familiar type name: `'42'`.

All you really have to do is change `valueType.TypeKind` to `valueType`. (It works because we overrode `ToString()` on `TypeSymbol`.)

Reply:
I did that at first then thought I should include the name of the type, like `string `or `integer `as opposed to `boolean`, so I went with the `valueType.TypeKind` :/ I agree that `42 `is better though - they'll know exactly where the error is coming from